src/position/decrease_position_collateral_utils.cairo

```diff
@@ -186,121 +708,83 @@ fn pay_for_cost(
-        collateral_token_price: price,
-        position_fee_factor: 0,
-        protocol_fee_amount: 0,
-        position_fee_receiver_factor: 0,
-        fee_receiver_amount: 0,
-        fee_amount_for_pool: 0,
-        position_fee_amount_for_pool: 0,
-        position_fee_amount: 0,
-        total_cost_amount_excluding_funding: 0,
-        total_cost_amount: 0,
-    };
-    (decrease_position_collateral_values, position_fees)
+    step: felt252
+) -> (position_utils::DecreasePositionCollateralValues, position_pricing_utils::PositionFees) {
+    if (!collateral_cache.is_insolvent_close_allowed) {
+        error::PositionError::INSUFFICIENT_FUNDS_TO_PAY_FOR_COSTS(
+            collateral_cache.result.remaining_cost_usd, step
+        );
+    }
+
+    params
+        .contracts
+        .event_emitter
+        .emit_position_fees_info(
+            params.order_key,
+            params.position_key,
+            params.market.market_token,
+            params.position.collateral_token,
+            params.order.size_delta_usd,
+            false, // isIncrease
+            fees
+        );
+
+    params
+        .contracts
+        .event_emitter
+        .emit_insolvent_close_info(
+            params.order_key,
+            params.position.collateral_amount,
+            *values.base_pnl_usd,
+            collateral_cache.result.remaining_cost_usd
+        );
+
+    (*values, get_empty_fees(@fees))
 }
 
 /// Return empty fees struct using fees struct given in parameter.
 /// Keep useful values such as accumulated funding fees.
 /// # Arguments
-/// * `fees` - The PositionFees struct used to get the new empty struct.
+/// * `fees` - The position_pricing_utils::PositionFees struct used to get the new empty struct.
 /// # Returns
-/// An empty PositionFees struct.
-fn get_empty_fees(fees: PositionFees) -> PositionFees {
-    // TODO
-    let address_zero = contract_address_const::<0>();
-    let position_referral_fees = PositionReferralFees {
-        referral_code: 0,
-        affiliate: address_zero,
-        trader: address_zero,
-        total_rebate_factor: 0,
-        trader_discount_factor: 0,
-        total_rebate_amount: 0,
-        trader_discount_amount: 0,
-        affiliate_reward_amount: 0,
-    };
-    let position_funding_fees = PositionFundingFees {
+/// An empty position_pricing_utils::PositionFees struct.
+fn get_empty_fees(
+    fees: @position_pricing_utils::PositionFees
+) -> position_pricing_utils::PositionFees {
+    let referral: position_pricing_utils::PositionReferralFees = Default::default();
+
+    // allow the accumulated funding fees to still be claimable
+    // return the latestFundingFeeAmountPerSize, latest_long_token_claimable_funding_amount_per_size,
+    // latest_short_token_claimable_funding_amount_per_size values as these may be used to update the
+    // position's values if the position will be partially closed
+    let funding = position_pricing_utils::PositionFundingFees {
```

Comment:
Try adding the type for that variable and all other variable where you did set the type

Reply:
I didn't put it here bc we see the type next to it

Reply:
Yes put I think its better to be explicit, we have nothing to lose